include/bitcoin/bitcoin/chain/chain_state.hpp

```diff
@@ -181,29 +184,39 @@ class BC_API chain_state
 
     static activations activation(const data& values, uint32_t forks);
     static uint32_t median_time_past(const data& values, uint32_t forks);
-    static uint32_t work_required(const data& values, uint32_t forks);
+    static uint32_t work_required(const data& values, uint32_t forks,
+        const settings& settings);
 
 private:
-    static size_t bits_count(size_t height, uint32_t forks);
+    static size_t bits_count(size_t height, uint32_t forks,
+        size_t retargeting_interval);
     static size_t version_count(size_t height, uint32_t forks);
     static size_t timestamp_count(size_t height, uint32_t forks);
-    static size_t retarget_height(size_t height, uint32_t forks);
+    static size_t retarget_height(size_t height, uint32_t forks,
+        size_t retargeting_interval);
     static size_t bip9_bit0_height(size_t height, uint32_t forks);
     static size_t bip9_bit1_height(size_t height, uint32_t forks);
 
-    static data to_pool(const chain_state& top);
+    static data to_pool(const chain_state& top, const settings& settings);
     static data to_block(const chain_state& pool, const block& block);
-    static data to_header(const chain_state& parent, const header& header);
+    data to_header(const chain_state& parent, const header& header,
+        const settings& settings);
 
-    static uint32_t work_required_retarget(const data& values);
-    static uint32_t retarget_timespan(const chain_state::data& values);
+    static uint32_t work_required_retarget(const data& values,
+        const settings& settings);
```

Comment:
Inconsistency here, pass `retarget_proof_of_work_limit`.

Reply:
If you check the implementation then you can see that actually four values from `settings` are used in `work_required_retarget()`:
* `settings.retarget_proof_of_work_limit`
* `settings.min_timespan`
* `settings.max_timespan`
* `settings.target_timespan_seconds`.
I'm happy to pass them individually but my judgement was that they would bloat the function signature. Your call.

Reply:
Yeah, these aren't always obvious calls. I've tried for a consistent approach whereby classes are constructed with the the settings object given sufficient use, but functions (including static members) are parameterized so as to avoid a dependency on the setting object.